src/libs/actions/Policy.js

```diff
@@ -78,35 +93,82 @@ function updateAllPolicies(policyCollection) {
 }
 
 /**
- * Fetches the policySummaryList from the API and saves a simplified version in Onyx
+ * Merges the passed in login into the specified policy
+ *
+ * @param {String} [name]
+ * @param {Boolean} [shouldAutomaticallyReroute]
+ * @returns {Promise}
  */
-function getPolicySummaries() {
-    API.GetPolicySummaryList()
-        .then((data) => {
-            if (data.jsonCode === 200) {
-                const policyDataToStore = _.reduce(data.policySummaryList, (memo, policy) => ({
-                    ...memo,
-                    [`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`]: getSimplifiedPolicyObject(policy),
-                }), {});
-                updateAllPolicies(policyDataToStore);
+function create(name = '', shouldAutomaticallyReroute = true) {
+    let res = null;
+    return API.Policy_Create({type: CONST.POLICY.TYPE.FREE, policyName: name})
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                // Show the user feedback
+                const errorMessage = translateLocal('workspace.new.genericFailureMessage');
+                Growl.error(errorMessage, 5000);
+                return;
+            }
+            res = response;
+
+            // We are awaiting this merge so that we can guarantee our policy is available to any React components connected to the policies collection before we navigate to a new route.
+            return Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${response.policyID}`, {
+                employeeList: getSimplifiedEmployeeList(response.policy.employeeList),
+                id: response.policyID,
+                type: response.policy.type,
+                name: response.policy.name,
+                role: CONST.POLICY.ROLE.ADMIN,
+            });
+        }).then(() => {
+            const policyID = lodashGet(res, 'policyID');
+            if (shouldAutomaticallyReroute) {
+                Navigation.dismissModal();
+                Navigation.navigate(policyID ? ROUTES.getWorkspaceCardRoute(policyID) : ROUTES.HOME);
             }
+            return Promise.resolve(policyID);
         });
 }
 
 /**
- * Fetches the policyList from the API and saves a simplified version in Onyx
+ * Fetches policy list from the API and saves a simplified version in Onyx, optionally creating a new policy first.
+ *
+ * More specifically, this action will:
+ * 1. Optionally create a new policy.
+ * 2. Fetch policy summaries.
+ * 3. Optionally navigate to the new policy.
```

Comment:
NAB, this makes it sound like there is another option to navigate to a created policy. But what actually happens is... if the policy is created we navigate to it.